File: back/src/main/java/csh/back/domain/vote/vote/dto/response/VoteFindListResponse.java
package csh.back.domain.vote.vote.dto.response;

import java.time.LocalDate;
import java.util.List;

public record VoteFindListResponse(
        LocalDate date,
        List<VoteWithTimelineResponse> timeLines
) {
        public static VoteFindListResponse of(LocalDate date, List<VoteWithTimelineResponse> timeLines) {
            return new VoteFindListResponse(date, timeLines);
        }
}

File: back/src/main/kotlin/csh/back/domain/vote/vote/dto/response/VoteFindListResponse.kt
package csh.back.domain.vote.vote.dto.response

import java.time.LocalDate

data class VoteFindListResponse(
    val date: LocalDate,
    val timeLines: List<VoteWithTimelineResponse>,
) {
    companion object {
        @JvmStatic
        fun of(date: LocalDate, timeLines: List<VoteWithTimelineResponse>): VoteFindListResponse =
            VoteFindListResponse(date, timeLines)
    }
}
